lib/optimize/ConcatenatedModule.js

```diff
@@ -999,7 +999,7 @@ class ConcatenatedModule extends Module {
 						const name = reference.identifier.name;
 						if (ConcatenationScope.isModuleReference(name)) {
 							const match = ConcatenationScope.matchModuleReference(name);
-							if (!match || match.ids.length < 1) continue;
```

Comment:
[0a8529f](https://github.com/webpack/webpack/pull/11751/commits/0a8529fcb18275f9d47b56e6ce43e6ed071bd185) reverts the changes in ConcatenatedModule.js from [95eabe4](https://github.com/webpack/webpack/pull/11751/commits/95eabe47ecf12e320c0733ae234dcf230e90e6c2) 
but im curious why this change fixes the initial bug?

Reply:
`findNewName` should already return a name that avoids collisions with other variables. But the actual bug was that in some cases potential collisions were not detected. This happened in the case when `ids = []`, which is the case for namespace objects or the externals (which are also treated as namespace objects).

The above snippet fixes the problem.